Formula/hdf5@1.10.rb

```diff
@@ -22,20 +21,20 @@ class Hdf5AT110 < Formula
   depends_on "automake" => :build
   depends_on "libtool" => :build
   depends_on "gcc" # for gfortran
-  depends_on "szip"
+  depends_on "libaec"
```

Comment:
Should have been changed in v1.10.7.
Unversioned `hdf5` dependency & dependents should be modified too.

https://portal.hdfgroup.org/display/HDF5/Szip+Compression+in+HDF+Products
> Please be aware that SZIP compression has been replaced by [AEC](https://gitlab.dkrz.de/k202009/libaec) compression as of HDF5-1.10.7. AEC is the open source version of SZIP.
>
> The AEC library can be obtained from the [German Climate Computing Center](https://www.dkrz.de/) from:
>
> https://gitlab.dkrz.de/k202009/libaec